tests/phpunit/plugin/ApiMapTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\plugin;
+
+use InvalidArgumentException;
+use PHPUnit\Framework\TestCase;
+use RuntimeException;
+use stdClass;
+
+/**
+ * @purpose (api map test purpose)
+ */
```

Comment:
```suggestion
```

Reply:
I'm not entirely sure what the point of this doc comment is, but using a test class to test itself is a bit dubious ...

Reply:
Again, I was just lazy to create another class. But this looks fine to me since ApiMap does not have any requirements on `$interface`. I would say no if this forces the class to implement a certain interface or something.

Reply:
I don't think this is really a significant issue. We could always change it to another class if we happen to have issues in the future, since this is not part of the API.